src/main/java/io/github/pascalklassen/pokefuture/PokemonService.java

```diff
@@ -230,29 +183,101 @@ private PokemonService() {
         return promise.future();
     }
 
-    public static void disableCashing(){
-        isCache = false;
+    public static void disableCaching() {
+        cachingEnabled = false;
+        invalidateAllCaches();
+        caches = null;
     }
 
-    public static void enableCashing(){
-        isCache = true;
+    public static void enableCaching() {
+        cachingEnabled = true;
+        invalidateAllCaches();
+    }
+
+    private static void invalidateAllCaches() {
+        if (caches != null) {
+            for (Class<?> clazz: caches.keySet()) {
```

Comment:
Du machst es hier was zu kompliziert, mach es doch lieber so:
```java
for (Cache cache : caches.values()) {
    invalidateCache(cache);
}
```

und dann übergibst du halt in der `invalidateCache`-Methode halt den Cache.